go-controller/pkg/ovn/acl.go

```diff
@@ -2,12 +2,64 @@ package ovn
 
 import (
 	"fmt"
+	knet "k8s.io/api/networking/v1"
+	"strings"
+
 	libovsdbclient "github.com/ovn-org/libovsdb/client"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
 )
 
+func joinACLName(substrings ...string) string {
+	return strings.Join(substrings, "_")
+}
+
+type ACLType string
+
+const (
+	ACLTypeIngress ACLType = "Ingress"
+	ACLTypeEgress  ACLType = "Egress"
+)
+
+func (aclType ACLType) ToPolicyType() knet.PolicyType {
+	if aclType == ACLTypeEgress {
+		return knet.PolicyTypeEgress
+	} else {
+		return knet.PolicyTypeIngress
+	}
+}
+
+// BuildACL should be used to build ACL instead of directly calling libovsdbops.BuildACL.
+// It can properly set and reset log settings for ACL based on ACLLoggingLevels
+func BuildACL(aclName string, priority int, match, action string,
+	logLevels *ACLLoggingLevels, aclType ACLType, externalIDs map[string]string) *nbdb.ACL {
+	var options map[string]string
+	var direction string
+	if aclType == ACLTypeIngress {
+		direction = nbdb.ACLDirectionToLport
+	} else {
+		direction = nbdb.ACLDirectionFromLport
+		options = map[string]string{
+			"apply-after-lb": "true",
```

Comment:
`apply-after-lb` was only being set for policy ACLs, right?
Do we know if this should be used generically for all types of ACLs?
Or is it a coincidence that we only use `ACLDirectionFromLport` from policy ACLs. What happens if we start to use `ACLDirectionFromLport` in other type of ACLs?
It looks to me like it should be kept specific to policy ACLs.

Reply:
that is a great question, let me try to explain
In ovn, there are 2 separate stages for `from-lport` acls denepding on `apply-after-lb`option
```
             OVN  will  apply the from-lport ACLs in two stages. ACLs without
              this option apply-after-lb set, will be applied before the  load
              balancer  stage  and  ACLs  with this option set will be applied
              after the load balancer stage.  The  priorities  are  indepedent
              between  these  stages  and may not be obvious to the CMS. Hence
              CMS should be extra careful when using this  option  and  should
              carefully  evaluate  the  priorities  of  all  the  ACLs and the
              default deny/allow ACLs if any.
```
it means that if we need to sort priorities for different egress ACLs, we need to make sure we use the same stage here. So if somebody wants to add new acls without this option it is similar to adding a new acl layer with its own priorities, which I don't have a use case for yet, but it will be a big change and may require a new `aclType`.

Reply:
Then I think that we either should not set `apply-after-lb` here, or if we want to map it to aclType, we should have three aclTypes from the get go, to avoid confusion
`aclTypeIngress, aclTypeEgressBeforeLB, aclTypeEgressAfterLB`
of two but like `aclTypeIngress, aclTypeEgressAfterLB`.